### PR TITLE
renamed envs

### PR DESCRIPTION
**Pull request by LianaHus**
fix https://github.com/ethereum/remix-project/issues/2538
fix https://github.com/ethereum/remix-project/issues/2431

### CODE REVIEW

**Comment by Aniket-Engg** [comment]
As we are looking to rename, we should consider this too: https://github.com/ethereum/remix-project/issues/925

**Comment by Aniket-Engg** [comment]
JavaScript VM -> Remix VM
Injected web3 -> Injected Provider
Web3 Provider -> External HTTP Provider

File: libs/remix-ui/run-tab/src/lib/reducers/runTab.ts [hunk 1]
```diff
@@ -118,27 +118,27 @@ export const runTabInitialState: RunTabState = {
...
       fork: 'berlin',
-      content: 'JavaScript VM (Berlin)'
+      content: 'Remix VM (Berlin)'
     }, {
       id: 'injected-mode',
       dataId: 'settingsInjectedMode',
       title: 'Execution environment has been provided by Metamask or similar provider.',
       value: 'injected',
-      content: 'Injected Web3'
+      content: 'Injected Provider - Metamask'
```
Comment (Aniket-Engg) [inline libs/remix-ui/run-tab/src/lib/reducers/runTab.ts:134]: Metamask should be conditional

**Review by Aniket-Engg — COMMENTED** [review]
Can you look for environment names like `Javascript VM`, `Injected Web3` in complete project and update it? It is there in Settings section and in recorder section.

File: libs/remix-ui/settings/src/lib/constants.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export const etherscanTokenTitle = 'EtherScan Access Token'
 export const etherscanTokenLink = 'https://etherscan.io/myapikey'
 export const etherscanAccessTokenText = 'Manage the api key used to interact with Etherscan.'
 export const etherscanAccessTokenText2 = 'Go to Etherscan api key page (link below) to create a new api key and save it in Remix.'
-export const ethereunVMText = 'Always use Javascript VM at load'
+export const ethereunVMText = 'Always use Remix VM at load'
 export const wordWrapText = 'Word wrap in editor'
 export const enablePersonalModeText = ' Enable Personal Mode for web3 provider. Transaction sent over Web3 will use the web3.personal API.\n'
```
Comment (Aniket-Engg) [inline libs/remix-ui/settings/src/lib/constants.ts:16]: web provider is still mentioned here

File: libs/remix-ui/helper/src/lib/components/web3Dialog.tsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ export function Web3ProviderDialog (props: web3ProviderDialogProps) {
         <br />For more info: <a href="https://remix-ide.readthedocs.io/en/latest/run.html#more-about-web3-provider" target="_blank" rel="noreferrer">Remix Docs on Web3 Provider</a>
```
Comment (Aniket-Engg) [inline libs/remix-ui/helper/src/lib/components/web3Dialog.tsx:27]: web3 provider is mentioned here too

**Review by Aniket-Engg — COMMENTED** [review]
I still see `Injected web3` and `web3 provider` mentioned at some places in the project, please consider updating them too

File: libs/remix-ui/run-tab/src/lib/reducers/runTab.ts [hunk 4]
```diff
@@ -118,27 +121,27 @@ export const runTabInitialState: RunTabState = {
...
       fork: 'berlin',
-      content: 'JavaScript VM (Berlin)'
+      content: 'Remix VM (Berlin)'
     }, {
       id: 'injected-mode',
       dataId: 'settingsInjectedMode',
       title: 'Execution environment has been provided by Metamask or similar provider.',
       value: 'injected',
-      content: 'Injected Web3'
+      content: `Injected Provider${(window && window.web3 && window.web3.currentProvider.isMetaMask) ? ' - Metamask' : ''}`
```
Comment (Aniket-Engg) [inline libs/remix-ui/run-tab/src/lib/reducers/runTab.ts:137]: window.ethereum is preferred over window.web3.currentProvider

**Review by yann300 — APPROVED** [review]

**Comment by fulldecent** [comment]
Nice, thank you!

**Pull request merged**